rest-api/dao/metrics_cache_dao.py

```diff
@@ -19,6 +20,66 @@
 class MetricsCacheJobStatusDao(UpdatableDao):
   def __init__(self):
     super(MetricsCacheJobStatusDao, self).__init__(MetricsCacheJobStatus)
+    self.test_hpo_id = HPODao().get_by_name(TEST_HPO_NAME).hpoId
+    self.test_email_pattern = TEST_EMAIL_PATTERN
+
+  def init_tmp_table(self):
+    with self.session() as session:
+      with warnings.catch_warnings():
+        warnings.simplefilter('ignore')
+        session.execute('DROP TABLE IF EXISTS metrics_tmp_participant;')
+      session.execute('CREATE TABLE metrics_tmp_participant LIKE participant_summary')
+
+      indexes_cursor = session.execute('SHOW INDEX FROM metrics_tmp_participant')
+      index_name_list = []
+      for index in indexes_cursor:
+        index_name_list.append(index[2])
+      index_name_list = list(set(index_name_list))
+
+      for index_name in index_name_list:
+        if index_name not in ['PRIMARY', 'participant_summary_hpo']:
+          session.execute('ALTER TABLE  metrics_tmp_participant DROP INDEX  {}'.format(index_name))
+
+      session.execute('CREATE INDEX idx_sign_up_time ON metrics_tmp_participant (sign_up_time)')
+      session.execute('CREATE INDEX idx_consent_time ON metrics_tmp_participant '
+                      '(consent_for_study_enrollment_time)')
+      session.execute('CREATE INDEX idx_member_time ON metrics_tmp_participant '
+                      '(enrollment_status_member_time)')
+      session.execute('CREATE INDEX idx_sample_time ON metrics_tmp_participant '
+                      '(enrollment_status_core_stored_sample_time)')
```

Comment:
Creating the indexes on the table before inserting the records will slow down the inserts.  Can the index creation lines be moved to after the data has been inserted?

Reply:
done